src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/message/EdifactMessage.java

```diff
@@ -0,0 +1,100 @@
+package uk.nhs.digital.nhsconnect.nhais.model.edifact.message;
+
+import java.util.Arrays;
+
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.DateTimePeriod;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.GpNameAndAddress;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.HealthAuthorityNameAndAddress;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.InterchangeHeader;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.MessageHeader;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionNumber;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionType;
+
+public class EdifactMessage {
```

Comment:
Should we agree some name for the entire EDIFACT file other than Message since this is a very overloaded term in our adaptor? EdifactFile? Within NHS are are treated as "files".

Reply:
Actually this abstraction is really representation of "EDIFACT message"

Reply:
Its an abstraction of the file, which contains an interchange and may contain one or more edifact messages

> UN/EDIFACT: a set of segments in the order specified in a message directory starting with the message header and ending with the message trailer.

https://www.unece.org/trade/untdid/texts/d300_s.htm#t

It would be nice if they had defined a term for the entire transmission, but they didn't :(

Reply:
For time being `MeshMessage` is "EDIFACT file" - when we implement multiple message handling the idea is that `Intercharge` will hold multiple `EdifactMessage` extracted from `MeshMessage`